### PR TITLE
added support for client-id header

### PR DESCRIPTION
**Pull request by lluft**

### CODE REVIEW

File: lib/castle-rb/client.rb [hunk 1]
```diff
@@ -42,6 +41,12 @@ def do_not_track?
 
     private
 
+    def get_client_id(request, response)
```
Comment (nijikon) [inline lib/castle-rb/client.rb]: `get_client_id` -> `client_id`

File: lib/castle-rb/client.rb [hunk 2]
```diff
@@ -51,6 +56,10 @@ def extract_cookie(request, response)
       end
     end
 
+    def get_client_id_header(request)
```
Comment (nijikon) [inline lib/castle-rb/client.rb]: `get_client_id_header` -> `client_id_from_header`

**Comment by bartes** [comment]
I think I reorganised that class a bit, but lets merge this one first.

**Comment by bartes** [comment]
we can close this in favour of https://github.com/castle/castle-ruby/pull/28

**Pull request closed without merging**